pkg/controller/job-controller.go

```diff
@@ -182,8 +199,25 @@ func (c *JobController) HandleObject(obj metav1.Object) {
 // SyncHandler compares the current Job state with the desired, and attempts to
 // converge the two. It then updates the Status block of the Job resource
 // with the current status of the resource.
-func (c *JobController) SyncHandler(key string) error {
-	klog.Info("Job Controller Loop!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
-
-	return nil
+func (c *JobController) SyncHandler(key string) (Result, error) {
+	klog.Info("Job Controller Loop!")
+	// Convert the namespace/name string into a distinct namespace and name
+	if key == "" {
+		runtime.HandleError(fmt.Errorf("Invalid resource key: %s", key))
+		return WrapResult(Result{}, nil)
+	}
+	namespace, tenantName := key2NamespaceName(key)
+	jobCR, err := c.minioClientSet.JobV1alpha1().MinIOJobs(namespace).Get(context.Background(), tenantName, metav1.GetOptions{})
+	if err != nil {
+		return WrapResult(Result{RequeueAfter: time.Second * 5}, nil)
+	}
+	// Loop through the different supported operations.
+	for _, val := range jobCR.Spec.Commands {
+		operation := val.Operation
+		if operation == "make-bucket" {
+			// TODO: Initiate a job to create the bucket(s) if they do not exist and if the Tenant is prepared for it.
+			klog.Info("This feature is work in progress...")
```

Comment:
I think the `// TODO:` is good enough for now, please remove this log line

Reply:
We have k8s client. I am follower of that.